src/java.desktop/share/classes/javax/swing/table/JTableHeader.java

```diff
@@ -1360,9 +1361,17 @@ public boolean contains(Point p) {
             }
 
             public Point getLocationOnScreen() {
+                Point parentLocation;
                 if (parent != null) {
-                    Point parentLocation = parent.getLocationOnScreen();
+                    try {
+                        parentLocation = parent.getLocationOnScreen();
+                    } catch (IllegalComponentStateException icse) {
```

Comment:
should we be preventing the exception by checking if the JTableHeader is visible or not instead?

Reply:
As per the spec, getLocationOnScreen() API can throw ICSE and it should be right to catch this exception rather than preventing it upfront by checking JTableHeader's visibility. Moreover, I tried checking with the visibility, still got the ICSE while performing the testing. So, I think this should be an ideal way to handle.

Reply:
Actually I was checking visibility with `parent.isVisible()` and this is always `true` but I was getting exception for getLocationOnScreen method. 
But when checked with `parent.isShowing()` and it behaves correctly, whenever it returns `false`, there is no exception as it doesn't go inside condition and if it returns `true`, `getLocationOnScreen` works correctly.

So, I guess `parent.isShowing` check should be sufficient to handle the exception.